packages/lib-classifier/src/components/Classifier/components/withKeyZoom/withKeyZoom.js

```diff
@@ -1,103 +1,24 @@
 import { MobXProviderContext } from 'mobx-react'
 import { Component, forwardRef } from 'react';
 
-function withKeyZoom (WrappedComponent) {
-  const ALLOWED_TAGS = ['svg', 'button', 'g', 'rect']
-
-  class KeyZoom extends Component {
-    constructor () {
-      super()
-      this.onKeyDown = this.onKeyDown.bind(this)
-    }
-
-    onKeyDown (e) {
-      const { classifierStore } = this.context
-      const {
-        panLeft,
-        panRight,
-        panUp,
-        panDown,
-        zoomIn,
-        zoomOut
-      } = classifierStore.subjectViewer
-      const htmlTag = e.target?.tagName.toLowerCase()
-      if (ALLOWED_TAGS.includes(htmlTag)) {
-        switch (e.key) {
-          case '+':
-          case '=': {
-            zoomIn()
-            return true
-          }
-          case '-':
-          case '_': {
-            zoomOut()
-            return true
-          }
-          case 'ArrowRight': {
-            e.preventDefault()
-            panRight()
-            return false
-          }
-          case 'ArrowLeft': {
-            e.preventDefault()
-            panLeft()
-            return false
-          }
-          case 'ArrowUp': {
-            e.preventDefault()
-            panUp()
-            return false
-          }
-          case 'ArrowDown': {
-            e.preventDefault()
-            panDown()
-            return false
-          }
-          default: {
-            return true
-          }
-        }
-      }
+import { useKeyZoom } from '@hooks'
 
-      return true
-    }
+const DEFAULT_HANDLER = () => true
 
-    render () {
-      const {
-        forwardedRef,
-        panLeft,
-        panRight,
-        panUp,
-        panDown,
-        zoomIn,
-        zoomOut,
-        ...props
-      } = this.props
-      return (
-        <WrappedComponent
-          ref={forwardedRef}
-          onKeyDown={this.onKeyDown}
-          {...props}
-        />
-      )
-    }
-  }
-
-  KeyZoom.contextType = MobXProviderContext
+function withKeyZoom (WrappedComponent) {
 
-  KeyZoom.defaultProps = {
-    forwardedRef: null,
-    panLeft: () => true,
-    panRight: () => true,
-    panUp: () => true,
-    panDown: () => true,
-    zoomIn: () => true,
-    zoomOut: () => true
+  function KeyZoom(props, ref) {
+    const { onKeyZoom } = useKeyZoom()
+    return (
+      <WrappedComponent
+        ref={ref}
+        onKeyDown={onKeyZoom}
+        {...props}
+      />
+    )
   }
 
-  const DecoratedKeyZoom = forwardRef(function (props, ref) {
-    return <KeyZoom {...props} forwardedRef={ref} />
-  })
+  const DecoratedKeyZoom = forwardRef(KeyZoom)
```

Comment:
Question: Why does withKeyZoom use `forwardRef`? I recently tried to find a case where `ref` was actually defined in a subject viewer component, but couldn't find a use case for a withKeyZoom ref anywhere in the classifier.

Reply:
If it won't break functionality, I suggest removing `forwardRef` from withKeyZoom component. In development mode locally, I often see errors in the console about incorrectly passing `ref` to functional components, and would like to cut down on unused `ref`s where we can.

Reply:
I've opened #4550 so that we can see if we use refs to the subject viewers anywhere. If it is being used, then React will error in the console about passing a ref to a functional component.

If you see that error on other components, the solution is to wrap the functional component in `forwardRef`.